Source/DafnyStandardLibraries/src/Std_Concurrent/Std_Concurrent.go

```diff
@@ -62,49 +64,41 @@ func (_this *MutableMap) Ctor__() {
 }
 func (_this *MutableMap) Keys() _dafny.Set {
   {
-    keys := make([]interface{}, 0)
-
-    _this.Internal.Range(func(key, value interface{}) bool {
-      keys = append(keys, key)
-      return true
-    })
-
-    return _dafny.SetOf(keys[:]...)
+    _this.mu.Lock()
+    keys := _this.Internal.Keys()
+    _this.mu.Unlock()
```

Comment:
This could be problematic where `.Keys()` throws an exception. In that case, the lock would never be released. Use `defer _this.mu.Unlock()` before calling the `.Keys()` or any other method if we have acquired a lock.

Reply:
Exactly why I asked you to review this, thank you. :) I noticed the rest of the file wasn't doing this either, will fix throughout.